### PR TITLE
Update Lucene to 8.9.0

### PR DESCRIPTION
**Pull request by mkleen**
https://lucene.apache.org/core/8_9_0/changes/Changes.html

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/engine/NoOpEngine.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ protected DirectoryReader open(final IndexCommit commit) throws IOException {
         final Directory directory = commit.getDirectory();
         final List<IndexCommit> indexCommits = DirectoryReader.listCommits(directory);
         final IndexCommit indexCommit = indexCommits.get(indexCommits.size() - 1);
-        return new DirectoryReader(directory, new LeafReader[0]) {
+        return new DirectoryReader(directory, new LeafReader[0], null) {
```
Comment (mkleen) [inline server/src/main/java/org/elasticsearch/index/engine/NoOpEngine.java:67]: See https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/engine/NoOpEngine.java#L65

**Comment by mkleen** [comment]
retest this please

**Comment by seut** [comment]
Our latest public release 4.5 contains Lucene 8.7.x so afaik we can't jump directly to 8.9.x within 4.6 as it would break compatibility. So we would have to defer this to 4.7.

**Comment by mkleen** [comment]
> Our latest public release 4.5 contains Lucene 8.7.x so afaik we can't jump directly to 8.9.x within 4.6 as it would break compatibility. We would have to defer this to 4.7.

I think 8.9.0 is fully backward compatibility with 8.7.x. as it is fully based on the 8.7 Codec https://lucene.apache.org/core/8_9_0/core/org/apache/lucene/codecs/lucene87/Lucene87Codec.html

**Comment by mkleen** [comment]
Minor releases in Lucene are always fully backward compatibility
https://cwiki.apache.org/confluence/display/LUCENE/BackwardsCompatibility

**Review by mfussenegger — APPROVED** [review]
:+1:

**Pull request merged**